From ceb97d1c8f5a456c5d49d3765eab85851ba28f2b Mon Sep 17 00:00:00 2001
From: MeiChia Chiu <MeiChia.Chiu@mediatek.com>
Date: Wed, 19 Nov 2025 15:34:00 +0800
Subject: [PATCH 092/105] mtk: wifi: mt76: mt7996: add support for emlsr
 enablement on one link

Register the capability of EMLSR enablement on one link

Signed-off-by: MeiChia Chiu <MeiChia.Chiu@mediatek.com>
---
 mt7996/init.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mt7996/init.c b/mt7996/init.c
index 853f53342..e97179492 100644
--- a/mt7996/init.c
+++ b/mt7996/init.c
@@ -118,6 +118,8 @@ static const struct wiphy_iftype_ext_capab iftypes_ext_capa[] = {
 			FIELD_PREP_CONST(IEEE80211_MLD_CAP_OP_MAX_SIMUL_LINKS,
 					 MT7996_MAX_RADIOS - 1) |
 			MT7996_NEG_TTLM_SUPPORT,
+		.ext_mld_capa_and_ops =
+			IEEE80211_EHT_ML_EXT_MLD_CAPA_EMLSR_ENA_ON_ONE_LINK,
 	}, {
 		.iftype = NL80211_IFTYPE_STATION,
 		.extended_capabilities = if_types_ext_capa_sta,
-- 
2.45.2

